lib/splitclient-rb/engine/matchers/in_list_semver_matcher.rb

```diff
@@ -10,20 +10,20 @@ def initialize(attribute, list_value, logger, validator)
       super(logger)
       @validator = validator
       @attribute = attribute
-      @semver_list = list_value.map { |item| SplitIoClient::Semver.new(item) }
+      @semver_list = list_value.map { |item| SplitIoClient::Semver.build(item, logger) }
```

Comment:
what happen if item is `null` or `empty` here ?

Reply:
In match function it checks if any of the items is nil and return false:
      value_to_match = SplitIoClient::Semver.build(args[:attributes][@attribute.to_sym], @logger)
      unless !value_to_match.nil? && @semver_list.all? { |n| !n.nil? }
        @logger.log_if_debug('whitelistMatcherData is required for IN_LIST_SEMVER matcher type')
        return false
      end